### PR TITLE
zookeeper: 3.7.2 -> 3.9.2

### PR DESCRIPTION
**Pull request by LeSuisse**
## Description of changes

Fixes CVE-2024-23944
https://www.openwall.com/lists/oss-security/2024/03/14/2

Release notes:
https://zookeeper.apache.org/doc/r3.9.2/releasenotes.html https://zookeeper.apache.org/doc/r3.8.0/releasenotes.html

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by LeSuisse** [comment]
Unless someone complains I'm likely to open a PR to mark the package with `knownVulnerabilities` on 23.11. The 3.7 branch is EOL for about a month and we use the binary tarball

File: pkgs/servers/zookeeper/default.nix [hunk 1]
```diff
@@ -47,8 +44,10 @@ stdenv.mkDerivation rec {
   meta = with lib; {
     homepage = "https://zookeeper.apache.org";
     description = "Apache Zookeeper";
+    changelog = "https://zookeeper.apache.org/doc/r${version}/releasenotes.html";
     license = licenses.asl20;
     maintainers = with maintainers; [ nathan-gs pradeepchhetri ztzg ];
     platforms = platforms.unix;
+    sourceProvenance = with sourceTypes; [ binaryNativeCode ];
```
Comment (risicle) [inline pkgs/servers/zookeeper/default.nix:51]: This should probably have `binaryBytecode` too.

**Review by risicle — APPROVED** [review]
`nixpkgs-review` happy, macos 12 x86_64 & nixos x86_64. `passthru.tests` pass, nixos x86_64

File: pkgs/servers/zookeeper/default.nix (hunk 1 above)
Comment (LeSuisse) [reply pkgs/servers/zookeeper/default.nix]: Adjusted, thanks! `binaryNativeCode` does not make sense here it's all `.jar`.

**Review by nathan-gs — APPROVED** [review]

File: pkgs/servers/zookeeper/default.nix (hunk 1 above)
Comment (risicle) [reply pkgs/servers/zookeeper/default.nix]: I did actually see something that looks like a copy of libressl (?) when unzipping the jar, complete with multiple architectures, so assumed you'd seen that.

**Pull request merged**